### PR TITLE
Deprecate debug options and always use query options

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Currently there are 2 ways to provide extra options when running a query: query options and debug options. Most of the options are provided through the query options, and user can embed query options into the query which makes it very easy to use. On the other hand, debug options can only be provided through the query parameter of the broker GET request, which is hard to use, and sometimes impossible when querying the controller. Managing 2 sets of query options also adds confusion and extra management overhead.
This PR simplifies the query options by also including the debug options into the query options, and only look at query options when executing the query.

## Release-Note
The following debug options can also be accepted as query options:
- routingOptions
- useScanReorderOpt

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]

**Review by siddharthteotia — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/startree/operator/StarTreeFilterOperator.java [hunk 1]
```diff
@@ -201,22 +197,22 @@ private BaseFilterOperator getFilterOperator() {
         int numPredicateEvaluators = predicateEvaluators.size();
         if (numPredicateEvaluators == 1) {
           // Single predicate evaluator
-          childFilterOperators
-              .add(FilterOperatorUtils.getLeafFilterOperator(predicateEvaluators.get(0), dataSource, numDocs));
+          childFilterOperators.add(
```
Comment (KKcorps) [inline pinot-core/src/main/java/org/apache/pinot/core/startree/operator/StarTreeFilterOperator.java:200]: nit: unnecessary formatting

**Review by KKcorps — APPROVED** [review]

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java [hunk 2]
```diff
@@ -1480,6 +1480,16 @@ private boolean forceLog(BrokerResponse brokerResponse, long totalTimeMs) {
   @VisibleForTesting
   static void setOptions(PinotQuery pinotQuery, long requestId, String query, JsonNode jsonRequest) {
     Map<String, String> queryOptions = new HashMap<>();
+    if (jsonRequest.has(Broker.Request.DEBUG_OPTIONS)) {
+      Map<String, String> debugOptions = getOptionsFromJson(jsonRequest, Broker.Request.DEBUG_OPTIONS);
+      if (!debugOptions.isEmpty()) {
+        LOGGER.debug("Debug options are set to: {} for request {}: {}", debugOptions, requestId, query);
+        pinotQuery.setDebugOptions(debugOptions);
```
Comment (KKcorps) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:1487]: We should not setDebugOptions at all IMO. Just putting it in query options ensures backward compatibility. We don't want other users/devs to rely on debug options for anything. Also `@Deprecated` annotation on `setDebugOptions`

Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:1487]: We do need to set it here for backward compatibility because the server might have not been upgraded to the newer version and is still expecting the option from the debug options. I didn't annotate `PinotQuery.setDebugOptions()` because it is auto-generated thrift class. Let me add some comments here to remove this after releasing `0.11.0`

File: pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentselector/RealtimeSegmentSelector.java [hunk 3]
```diff
@@ -95,8 +93,8 @@ public void onAssignmentChange(IdealState idealState, ExternalView externalView,
         if (instanceStateMap.containsValue(SegmentStateModel.CONSUMING)) {
           // Keep the first CONSUMING segment for each partition
           LLCSegmentName llcSegmentName = new LLCSegmentName(segment);
-          partitionIdToFirstConsumingLLCSegmentMap
-              .compute(llcSegmentName.getPartitionGroupId(), (k, consumingSegment) -> {
+          partitionIdToFirstConsumingLLCSegmentMap.compute(llcSegmentName.getPartitionGroupId(),
```
[the remark this answers is not in the document]
Comment (Jackie-Jiang) [reply pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentselector/RealtimeSegmentSelector.java:96]: My IDE auto-reformatted this based on the Pinot Style, but since it is not relevant, reverting it

File: pinot-core/src/main/java/org/apache/pinot/core/startree/operator/StarTreeFilterOperator.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/startree/operator/StarTreeFilterOperator.java:200]: Reverted
